### DISCUSSION TITLE
Why `Button.on_click()` was deprecated?

### DISCUSSION THREAD

**Discussion by mattpap** [bokeh/bokeh#12360] [Internals and design (Q&A)]
Currently I get:
```
BokehDeprecationWarning: on_click(handler) was deprecated in Bokeh 3.0.0 and will be removed, use on_event("button_click", handler) instead.
```
This change seems to me counter productive. I also noticed that `on_click()` is used a lot, at least in submitted bug reports (as I've been testing dozens of issues for PR #11915).

**Comment by bryevdv** [comment]
I don't recall but I'm not opposed to removing true deprecation since dask uses it.

**Comment by bryevdv** [comment]
@mattpap here is the relevant PR, I have not had a chance to dig into the details. There was an underlying concern, and the PR did more than just adding the deprecations, so perhaps just undoing the deprecation parts is fine

* https://github.com/bokeh/bokeh/pull/11942

**Reply by bryevdv** [reply]
Here is a tentative PR to undeprecate: https://github.com/bokeh/bokeh/pull/12372
